pubsub/pubsub.go

```diff
@@ -255,12 +286,31 @@ func (s *Subscription) Receive(ctx context.Context) (_ *Message, err error) {
 		}
 		// No messages are available and there are no calls to ReceiveBatch in flight.
 		// Make a call.
+		//
+		// Ask for a number of messages that will give us the desired queue length.
+		// Unless we don't have information about process time (at the beginning), in
+		// which case just get one message.
+		nMessages := 1
```

Comment:
Why 1? Seems like a very conservative estimate. It "guesses" that it will take 2 seconds to process a message.

Reply:
2 or more seconds. Say it takes three minutes to process a message. Then we never want to fetch more than one at a time. If we started by fetching two, then the second would lie around for a while and might expire. 

I think it's better to be conservative at the beginning. Applications with long processing times will be happy that their messages don't expire. Everyone else will ramp up very quickly (since after that first message is serviced in 10ms, the next ReceiveBatch call will ask for 200 messages).